cylc/flow/data_store_mgr.py

```diff
@@ -311,17 +336,23 @@ def apply_delta(key, delta, data):
             for field in CLEAR_FIELD_MAP[key]:
                 if field in field_set or delta.updated.states_updated:
                     data[key].ClearField(field)
+
+            # Apply the updated delta
             data[key].MergeFrom(delta.updated)
+
+            # Deque (i.e. length restrict) selected fields
+            for field_name, max_len in DEQUE_FIELD_MAP.get(key, {}).items():
+                if field_name in field_set:
+                    lst = getattr(data[key], field_name)
+                    while len(lst) > max_len:
+                        lst.pop(0)
```

Comment:
This block will ensure the data store only holds the last 10 log messages.

Reply:
Consider adding to code comment?